### PR TITLE
Fix JsonObject type guard

### PR DESCRIPTION
**Pull request by slifty**
This PR fixes an edge case where `isJsonObject` would return true for `null` values.  It also adds unit tests for good measure.

### CODE REVIEW

File: src/types/JsonObject.ts [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
+// in the swagger-ui-express library.  Currently that definition is:
+//
+// export interface JsonObject {
+//     [key: string]: any;
+// }
+//
+// ^ this is essentially the same as the definition of a non-null Object
 export const isJsonObject = (obj: unknown): obj is JsonObject => (
   typeof obj === 'object'
+  && obj !== null
```
Comment (bickelj) [inline src/types/JsonObject.ts:13]: This is still disconnected from the actual definition of `JsonObject` right? If it works, that's fine, but we're basically defining our own `JsonObject` rather than using an existing `JsonObject`, right?

Comment (bickelj) [reply src/types/JsonObject.ts:13]: Suppose the definition within `swagger-ui-express` changes. Now we have a separate definition of our own. I suppose an error at runtime would occur. But then in that case what was the point of the type checking?

Comment (bickelj) [reply src/types/JsonObject.ts:13]: I now see your comments in the #71 area discussing the same. My summary: library user type guards are a necessary hack for libraries that do not supply them. If the type guard were not necessary there would be no disconnect. So whenever we introduce a type guard for a library this same disconnect happens. Close?

Comment (slifty) [reply src/types/JsonObject.ts:13]: Want to be sure you saw my explanation over in that other thread!

Comment (bickelj) [reply src/types/JsonObject.ts:13]: The question is less important for this simple instance and more important for my understanding or a more complex/impactful instance. I am satisfied with your explanation even if I keep treating TypeScript like a monkey with a hammer.

**Review by bickelj — APPROVED** [review]

File: src/types/JsonObject.ts (hunk 1 above)
Comment (slifty) [reply src/types/JsonObject.ts:13]: I definitely want to make sure you walk away understanding -- this may be an "ah HA!" moment once we're through it!

That said, your question did get me to do the more correct thing and lean into ajv for this (that way if the definition changes AJV will let us know right away in the form of a static analysis error).
